stock_location_analytic_purchase/views/purchase_views.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="purchase_order_form" model="ir.ui.view">
+        <field name="name">purchase.order.form</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form"/>
+        <field name="arch" type="xml">
+            <field name="dest_address_id" position="after">
+                <field name="check_analytic_id" invisible="1"/>
+            </field>
+            <field name="dest_address_id" position="attributes">
+                <attribute name="attrs">{'required': [('check_analytic_id', '==', 'True')]}</attribute>
```

Comment:
It should be not required when the analytic account is set. That means I'm moving to a location inside the company, thus, it is not a drop ship case. Moreover, this is overwriting completely the standard attributes. In order to avoid that you should include the attribute as it is in the standard plus the new thing:
{'invisible': ['|',('default_location_dest_id_usage', '!=', 'customer'), ('check_analytic_id', '==', true)], 'required': ['|',('default_location_dest_id_usage', '=', 'customer'),('check_analytic_id', '==', false)]}